src/FileInput/FileInput.jsx

```diff
@@ -0,0 +1,136 @@
+      </g>
+    </g>
+  </svg>);
+}
+
+UploadingIcon.propTypes = {
+  percent: React.PropTypes.number.isRequired,
+};
+
+export class FileInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      filename: null,
+      error: null,
+      progress: null,
+      success: null,
+    };
+    this.onDropRejected = this.onDropRejected.bind(this);
+    this.onDropAccepted = this.onDropAccepted.bind(this);
+  }
+
+  onDropAccepted(files) {
+    const fileKey = Math.random();
+    this.setState({fileKey});
+    this.setState({filename: files[0].name, error: null, progress: 0, success: null});
+    this.props.store(files[0], {
+      success: () => {
+        if (this.state.fileKey !== fileKey) {
+          return; // If the user drops multiple files we want to only treat the last one as "success"
+        }
+        this.setState({success: true});
+      },
+      error: () => {
+        this.setState({error: "Unknown error"});
+      },
+      progress: (progress) => {
+        if (this.state.fileKey !== fileKey) {
+          return; // If the user drops multiple files we want to only take progress from the active upload
+        }
+        this.setState({progress});
+      },
+    });
+  }
+
+  onDropRejected() {
+    this.setState({filename: null, error: `Invalid ${this.props.label}`, success: null});
+  }
+
+  render() {
+    return (<Dropzone
+      accept={this.props.accept}
+      style={{}}
+      multiple={false}
+      onDropAccepted={this.onDropAccepted}
+      onDropRejected={this.onDropRejected}
+    >{({isDragActive, isDragReject}) => {
+      let message = `Upload ${this.props.label}`;
+      let icon = <DefaultIcon />;
+      let selected = false;
+      if (isDragActive) {
+        selected = true;
+        message = "";
+      } else if (isDragReject || this.state.error) {
+        selected = true;
+        message = this.state.error || `Invalid ${this.props.label}`;
+        icon = <ErrorIcon />;
+      } else if (this.state.success) {
+        selected = true;
+        message = this.state.filename;
+        icon = <SuccessIcon />;
+      } else if (this.state.filename) {
+        icon = <UploadingIcon percent={this.state.progress} />;
+        message = this.state.filename;
+        selected = true;
+      }
+      return (<FlexBox className="FileInput" grow >
+        {selected && <label className="FileInput--Label">{this.props.label}</label>}
+        <FlexItem className={`FileInput--Text ${selected ? "FileInput--Text--selected" : ""}`} grow>{message}</FlexItem>
```

Comment:
Can you just put `{message}` on a separate line to reduce the width of the line? easier to read on narrower windows like my editor/Github